src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDocumentClassifierPass.cs

```diff
@@ -79,6 +79,8 @@ protected override CodeTarget CreateTarget(RazorCodeDocument codeDocument, Razor
             computedClass = ComponentMetadata.MangleClassName(computedClass);
         }
 
+        @class.Annotations[CommonAnnotations.NullableContext] = CommonAnnotations.NullableContext;
```

Comment:
Frankly, this concerns me a bit. How does a user opt-out of this? This can cause new nullable warnings in user code or possibly make a type a stronger guarantee than a user was intending.

Reply:
That should be handled correctly since it adds `#nullability restore`, not `enable`. If the user has nullability disabled, it won't have effect. For example, if I don't enable nullability in the added test, I get a warning on the `@inherits BaseComponent<string?>`:

<samp>
warning CS8669: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. Auto-generated code requires an explicit '#nullable' directive in source.
</samp>

Reply:
Still, though, it's a potentially breaking change for users, as it will change the meaning of existing ocde. @jaredpar, do we have any bar or process for this type of change?

Reply:
I'm a bit confused on whath the behavior change is here. Can we summarize? 

Particularly what are the scenarios where customers would get a new warning on upgrade?

Reply:
Consider the added test in `src/Compiler/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentCodeGenerationTestBase.cs`, but with a slight modification: instead of `string?`, it's `string`. Today, that `string` would be an oblivious value: after this change, it's a non-null value. The user couldn't have used `string?` there before, so their component could now be stating a stronger generic type (`string!`) where it was not intended, as it was previously `string~`.

Reply:
It's potentially breaking, but it doesn't create errors, only warnings. Plus it fixes a highly-upvoted bug.

Reply:
Btw, seems that https://github.com/dotnet/razor/pull/8909 introduced similar change for type arguments and https://github.com/dotnet/razor/pull/8825 for type constraints.

Previously:

```cs
public partial class Component1<TItem> where TItem : string
```

Is now generated as:

```cs
public partial class Component1<
#nullable restore
#line 1 "...\Component1.razor"
TItem

#line default
#line hidden
#nullable disable
    >
#nullable restore
#line 1 "...\Component1.razor"
where TItem1 : string

#line default
#line hidden
#nullable disable
```

So that's a similar break, I think.